src/HasteModuleLoader/HasteModuleLoader.js

```diff
@@ -226,9 +226,12 @@ Loader.prototype._execModule = function(moduleObj) {
     || (onlyCollectFrom && onlyCollectFrom[modulePath] === true);
 
   if (shouldCollectCoverage) {
+    if (!this._CoverageCollector) {
+      this._CoverageCollector = require(this._config.coverageCollector);
+    }
```

Comment:
Can you explain what you mean that the require in the constructor is not the real one? It should be the same one you get here...